packages/core/src/components/forms/inputGroup.tsx

```diff
@@ -91,6 +91,15 @@ export class InputGroup extends React.Component<HTMLInputProps & IInputGroupProp
         this.updateInputWidth();
     }
 
+    private maybeRenderLeftIcon() {
+        const { leftIconName } = this.props;
+        if (leftIconName == null) {
+            return null;
+        }
+        const iconClass = Classes.iconClass(leftIconName);
```

Comment:
I like to separate things out to make each line as dumb (and therefore as glanceable) as possible.